fedimint-api/src/db/mod.rs

```diff
@@ -879,4 +1031,79 @@ mod tests {
 
         assert_eq!(returned_keys, expected_keys);
     }
+
+    pub async fn verify_simple_migration(old_db: Database, new_db: Database) {
+        let mut dbtx = old_db
+            .begin_transaction(ModuleDecoderRegistry::default())
+            .await;
+
+        assert!(dbtx
+            .insert_entry(&DatabaseVersionKey, &"1.0.0".to_string())
+            .await
+            .is_ok());
+
+        let mut i = 0;
+        while i < 100 {
+            assert!(dbtx
+                .insert_entry(&TestKey { value: i + 100 }, &TestVal(i + 101))
+                .await
+                .is_ok());
+            i = i + 1;
+        }
+
+        dbtx.commit_tx().await.expect("DB Error");
+
+        // Upgrade fedimint
+        let mut dbtx = new_db
+            .begin_transaction(ModuleDecoderRegistry::default())
+            .await;
+        assert!(dbtx
+            .insert_entry(&DatabaseVersionKey, &"1.0.1".to_string())
+            .await
+            .is_ok());
+        dbtx.commit_tx().await.expect("DB Error");
+
+        let mut migrator = DatabaseMigrator {
+            source: old_db.clone(),
+            target: new_db.clone(),
+            migrations: BTreeMap::new(),
+        };
+        migrator.add_migration(
+            "1.0.0".to_string(),
+            Box::new(|source, target| async move {
+                let mut source_dbtx = source
+                    .begin_transaction(ModuleDecoderRegistry::default())
+                    .await;
+                let mut target_dbtx = target
+                    .begin_transaction(ModuleDecoderRegistry::default())
+                    .await;
+                migrate_prefix!(
+                    source_dbtx,
+                    target_dbtx,
+                    DbPrefixTestPrefix,
+                    TestKeyV1,
+                    TestVal
+                );
+                target_dbtx.commit_tx().await.expect("DB Error");
+            }),
+        );
+        migrator.migrate("1.0.1".to_string()).await;
```

Comment:
Key/value pairs that don't have a migration currently will be dropped. Need to fix :)